From 45b401a947af944c20b3c451a35dfe53bca5ef3b Mon Sep 17 00:00:00 2001
From: Tim Orling <timothy.t.orling@linux.intel.com>
Date: Thu, 28 Dec 2017 21:24:57 -0800
Subject: [PATCH 1/3] comparison_fn_t is glibc specific, use raw signature in
 function pointer

Make it work with musl where comparison_fn_t is not provided

Reuse the approach from systemd:
systemd/0013-comparison_fn_t-is-glibc-specific-use-raw-signature-.patch

Based on work by: Khem Raj <raj.khem@gmail.com>

Signed-off-by: Tim Orling <timothy.t.orling@linux.intel.com>
Upstream-Status: Submitted [https://github.com/systemd/systemd-bootchart/pull/47]
---

 src/util.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/util.h b/src/util.h
index 561f9e7..e9cfd81 100644
--- a/src/util.h
+++ b/src/util.h
@@ -57,7 +57,7 @@ extern char **saved_argv;
  * Normal qsort requires base to be nonnull. Here were require
  * that only if nmemb > 0.
  */
-static inline void qsort_safe(void *base, size_t nmemb, size_t size, comparison_fn_t compar) {
+static inline void qsort_safe(void *base, size_t nmemb, size_t size, int (*compar)(const void *, const void *)) {
         if (nmemb <= 1)
                 return;
 
-- 
2.13.6

